migrator/migrations/m_183_to_m_184_create_declarative_config_health_table/migration.go

```diff
@@ -0,0 +1,34 @@
+package m183tom184
```

Comment:
Is the migration only creating the table and not populating it from some other source?  We only need a migration if we are going to populate the data.  The table resources should be created without the need for a migration.  Obviously you should run an upgrade to confirm that.

Follow up question, where is the data now and does the data need to be moved into this new table?

Reply:
This migration only creates the table - we took inspiration from this PR which adds similar migration for notificationschedules table: https://github.com/stackrox/stackrox/pull/5067

However, we would need to migrate data from `integrationhealth` store to this one(more specifically, all objects with `DECLARATIVE_CONFIG` type).

Currently, I'm a little bit unsure how it is supposed to work - I see a few options:
1. We create two separate migrations - `create table` migration and `move data to table` migration
2. We create one migration - `create table + move data to table` migration
3. `create table` migration is not needed, only `move data to table` migration

Reply:
Ah, well my comment there applies.  But since then my thinking has evolved.  

Option 3 is not valid because what postgres things that aren't created by a migration are not created until after the migrations.  My preference is option 2 for a couple of reasons.  It reduces our migrations and in reality it doesn't matter whether the table is created or not until you try to use it.

For my own education, declarative config was added in 4.0?

Reply:
>For my own education, declarative config was added in 4.0?
It was added as a developer preview, so I guess we don't have to be as strict here w.r.t notifying people in the CHANGELOG or something like this. 4.1 will be the GA release of declarative config.
 
Just to know this myself: the existing migrations we have where we added empty tables essentially should have never been migrations in the first place, right?

Reply:
> Would there potentially be any declarative config data existing in customer environments?

Yes. However, AFAIU this functionality was presented only to a few customers who expressed interest in experimenting with this feature

Reply:
got it.  To be safe then when you hook up the services to the store you should probably migrate the data